src/components/MonthCalendar/index.js

```diff
@@ -0,0 +1,170 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { useUniqueIdentifier, useLocale } from '../../libs/hooks';
+import { getFormattedMonth, getYearsRange, getLastDayMonth, addMonths } from '../Calendar/helpers';
+import getCurrentMonth from './helpers/getCurrentMonth';
+import { Provider } from './context';
+import Select from '../Select';
+import RightIcon from './icons/rightArrow';
+import LeftIcon from './icons/leftArrow';
+import DaysOfWeek from './daysOfWeek';
+import Month from './month';
+import StyledControlsContainer from './styled/controlsContainer';
+import StyledMonthContainer from './styled/monthContainer';
+import StyledArrowButton from './styled/arrowButton';
+import StyledMonth from './styled/month';
+import StyledTable from './styled/table';
+
+export default function MonthCalendar(props) {
+    const {
+        id,
+        className,
+        style,
+        minDate,
+        maxDate,
+        currentMonth,
+        selectedDate,
+        locale,
+        onSelectDate,
+        onMonthChanged,
+        dateComponent,
+    } = props;
+    const monthLabelId = useUniqueIdentifier('month');
+    const currentLocale = useLocale(locale);
+    const month = useMemo(() => getCurrentMonth(currentMonth, minDate, maxDate), [
+        currentMonth,
+        minDate,
+        maxDate,
+    ]);
+    const currentYear = month.getFullYear();
+    const yearsRange = useMemo(
```

Comment:
can we abstract these hooks like
```js
const yearRange = getYearRange(minDate, maxDate, month);
```

then different file in `./hooks/getYearRange.js`
```js
const getYearRange = (minDate, maxDate, month) => {
    return useMemo(() => {
         .... 
    }, [minDate, maxDate, month])
}
```